test/e2e/specs/fakerp/metricsbridge.go

```diff
@@ -0,0 +1,133 @@
+		"kubelet_runtime_operations_errors != 0": true,
+		"kubelet_docker_operations_errors != 0":  true,
+	}
+
+	BeforeEach(func() {
+	})
+
+	It("should be possible to get data for each metric in defined queries", func() {
+		By("getting the configMap")
+		mbConfigMap, err := sanity.Checker.Client.Admin.CoreV1.ConfigMaps("openshift-azure-monitoring").Get("metrics-bridge", meta_v1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		var mbConfig struct {
+			Interval           time.Duration `json:"intervalNanoseconds,omitempty"`
+			PrometheusEndpoint string        `json:"prometheusEndpoint,omitempty"`
+			StatsdSocket       string        `json:"statsdSocket,omitempty"`
+
+			Queries []struct {
+				Name  string `json:"name,omitempty"`
+				Query string `json:"query,omitempty"`
+			} `json:"queries,omitempty"`
+
+			Account   string `json:"account,omitempty"`
+			Namespace string `json:"namespace,omitempty"`
+
+			Region            string `json:"region,omitempty"`
+			SubscriptionID    string `json:"subscriptionId,omitempty"`
+			ResourceGroupName string `json:"resourceGroupName,omitempty"`
+			ResourceName      string `json:"resourceName,omitempty"`
+
+			Token              string `json:"token,omitempty"`
+			InsecureSkipVerify bool   `json:"insecureSkipVerify,omitempty"`
+
+			log        *logrus.Entry
+			rootCAs    *x509.CertPool
+			prometheus v1.API
+			rt         http.RoundTripper
+			conn       net.Conn
+		}
+
+		By("unmarshalling the ConfigMap")
+		err = yaml.Unmarshal([]byte(mbConfigMap.Data["config.yaml"]), &mbConfig)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(mbConfig.Queries)).NotTo(BeZero())
+
+		By("getting the Prometheus token and URL")
+		token, err := sanity.Checker.Client.Admin.GetServiceAccountToken("openshift-monitoring", "prometheus-k8s")
+		Expect(err).NotTo(HaveOccurred())
+
+		route, err := sanity.Checker.Client.Admin.RouteV1.Routes("openshift-monitoring").Get("prometheus-k8s", meta_v1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("dialing Prometheus")
+		cli, err := api.NewClient(api.Config{
+			Address: "https://" + route.Spec.Host,
+			RoundTripper: &authorizingRoundTripper{
+				RoundTripper: &http.Transport{
+					TLSClientConfig: &tls.Config{
+						RootCAs:            nil,
+						InsecureSkipVerify: true,
+					},
+				},
+				token: string(token),
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		prometheusApi := v1.NewAPI(cli)
+
+		for _, q := range mbConfig.Queries {
+			By(fmt.Sprintf("checking query %s", q.Query))
+			//checking if the whole query works
+			value, err := prometheusApi.Query(context.Background(), q.Query, time.Time{})
+			Expect(err).NotTo(HaveOccurred())
+			if len(value.String()) == 0 {
+				//if there are no datapoints in the literar query, check if it's by design
+				_, present := manuallyVerifiedQueries[q.Query]
+				if present {
+					//query verified manually
+					continue
+				}
+
+				re := regexp.MustCompile(`(.+)\s*!=\s*0`)  // if ends with != 0
```

Comment:
consider moving this and the following comments into a comment section above this test spec which documents the entire flow logic for this test (as you explained to me :) )

Reply:
done.